### PR TITLE
[WIP] Initial setup for system check command

### LINKED ISSUE #1754: [mesheryctl] Create `mesheryctl system check`
Child Issue- [mesheryctl] [child] Create mesheryctl exp preflight to verify environment readiness for a Meshery deloyment #2607

**Current Behavior**
Users are able to run `mesheryctl system status` to check whether Meshery containers are running or not in their local Docker environment (using docker-compose). Users are not, however, able to perform to other types of checks in-depth:

1.  `check` - a sophisticated check encompassing current environment and current Meshery deployment health. Should include aspects of `system version` and of `system status`.
1. `preflight` - for verification of presence of dependencies and overall environment readiness for a Meshery deployment

**Desired Behavior**
`mesheryctl` should provide a utility to aid in the troubleshooting of Meshery deployments. This utility will need to run a number of health checks.

**Implementation**
1. As a subcommand of the `experimental` command, create `mesheryctl experimental system check`.
1. Invoke the collection of connectivity tests between Meshery server, its adapters, Kubernetes, and any connected add-ons like Grafana, Prometheus, Kubernetes permission verification, and Meshery version check.
1. This utility should support a `--preflight` flag like so: `mesheryctl system check --preflight`, which only runs the Docker and Kubernetes checks.

**Mockups**
```
➜  ~ mesheryctl system check

Docker
--------------
√ Docker is running
√ docker-compose is available

Kubernetes API
--------------
√ can initialize the client
√ can query the Kubernetes API
√ has required level of privileges

Kubernetes Version
------------------
√ is running the minimum Kubernetes API version
√ is running the minimum kubectl version

Meshery Version
---------------
√ server is up-to-date (stable-v0.x.x)
√ cli is up-to-date (stable-v0.x.x)

Meshery Adapters
---------------
√ meshery-istio adapter is reachable and running
√ meshery-consul adapter is reachable and running
√ meshery-linkerd adapter is reachable and running
.
.
.
√ meshery-nginx-sm adapter is reachable and running
√ meshery-octarine adapter is reachable and running
```

**Acceptance Tests**
1. _Meshery Preflight_
  - Users may or may not be installing to Docker, so failure of Docker being available should only be shown as a warning.

2. _Meshery Version_ 
  - To verify the Meshery server version and the Meshery client version, this check should first attempt to use the `/api/server/version` endpoint on Meshery server and should use the `mesheryctl system version` function to check on the client version. It should inform the user if a later version of either server or client is available. It should warn the user if the server and client are not at the same version number.

3. _Meshery Adapters_
  - To verify availability of Meshery adapters, this check should invoke the [`/api/mesh/adapter/ping` endpoint](https://github.com/layer5io/meshery/blob/master/handlers/mesh_ops_handlers.go#L268) on Meshery to run an ad-hoc connectivity tests for all known adapters.
  - A user may have disabled some or all of Meshery's adapters. Irrespective, all known adapters should be checked for availability. 

4. _Kubernetes_
  - Meshery requires "ClusterAdmin" role in order to be able to provide and manage service meshes. This check should verify that the Meshery ServiceAccount has the "ClusterAdmin" role assigned.

### PR DESCRIPTION
**Pull request by nitishm**
Signed-off-by: Nitish Malhotra <nitish.malhotra@gmail.com>

**Description**

This PR fixes #1754 

**Notes for Reviewers**

This is a new command to be added under the `mesheryctl exp` subcommand tree that checks the status of all meshery components and the dependencies, environments, etc.

Changes made only to `mesheryctl` package.

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [X] Yes, I signed my commits.

### CODE REVIEW

File: mesheryctl/internal/cli/root/experimental/system/system.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	"fmt"
+
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+const (
+	url     = "http://localhost:9081"
+	fileURL = "https://raw.githubusercontent.com/layer5io/meshery/master/docker-compose.yaml"
```
Comment (kushthedude) [inline mesheryctl/internal/cli/root/experimental/system/system.go:27]: Is it being used somewhere?

**Review by kushthedude — COMMENTED** [review]
Looks good to me for the preFlight check, just a suggestion can we rename the preflight flag to pre?

File: mesheryctl/internal/cli/root/experimental/system/system.go (hunk 1 above)
Comment (nitishm) [reply mesheryctl/internal/cli/root/experimental/system/system.go:27]: Don't think so. Will remove it.

**Comment by nitishm** [comment]
> Looks good to me for the preFlight check, just a suggestion can we rename the preflight flag to pre?

Sure good idea. Less verbose.

**Comment by leecalcote** [comment]
Hmpf, I missed this PR earlier. What a great use of the `exp` area. There's lint check going on about whitespace formatting, but otherwise good to go here. 👍

**Review by leecalcote — APPROVED** [review]

File: mesheryctl/internal/cli/root/experimental/system/check.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var (
+	preflightChecks bool = false
+)
+
+// StopCmd represents the stop command
+var checkCmd = &cobra.Command{
```
Comment (subhamkrai) [inline mesheryctl/internal/cli/root/experimental/system/check.go:33]: `StopCmd` `checkCmd`? maybe you need to update the comment

**Pull request closed without merging**